airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -3021,7 +3021,15 @@ def execute(self, context: Any):
         # Wait for the job to complete
         if not self.deferrable:
             job.result(timeout=self.result_timeout, retry=self.result_retry)
-            self._handle_job_error(job)
+            while True:
```

Comment:
I'd like to hear the Google team's opinion, but having another look at [QueryJob's result method](https://github.com/googleapis/python-bigquery/blob/ba61a8ab0da541ba1940211875d7ea2e9e17dfa8/google/cloud/bigquery/job/query.py#L1673-L1690), I'm not sure this is even necessary, because `result` now has logic to determine whether or not a job is `DONE`.